MDANSE_GUI/Src/MDANSE_GUI/InputWidgets/TrajectoryFilterWidget.py

```
@@ -0,0 +1,1463 @@
+from MDANSE.Framework.Configurators.TrajectoryFilterConfigurator import (
+    TrajectoryFilterConfigurator,
+)
+from MDANSE.Mathematics.Signal import (
+    Filter,
+    filter_description_string,
+    FILTER_MAP,
+    DEFAULT_FILTER_CUTOFF,
+    DEFAULT_N_STEPS,
+    DEFAULT_TIME_STEP,
+    power_spectrum,
+)
+import matplotlib.pyplot as mpl
+from matplotlib.backends.backend_qt5agg import FigureCanvasQTAgg
+from matplotlib.backends.backend_qt5agg import (
+    NavigationToolbar2QT as NavigationToolbar2QTAgg,
+)
+
+# Default maximum value for a float spinbox
+DEFAULT_SPINBOX_MAX_FLOAT = 1000.0
+
+# Default step size for a float spinbox
+DEFAULT_SPINBOX_STEP_FLOAT = 0.1
+
+# Decimal precision for a float spinbox
+FLOAT_SPINBOX_DECIMALS = 4
+
+
+class ConstrainedDoubleSpinBox(QDoubleSpinBox):
+    """Custom QDoubleSpinBox allowing for the application of a function
+    to changed values in order to find an appropriate new value that satisfies the constraint.
+
+    """
+
+    def __init__(self, minimum: float, maximum: float, step: float, value: float):
+        """
+        Parameters
+        ----------
+        minimum: float
+            Spinbox minimum value.
+        maximum: float
+            Spinbox maximum value.
+        step: float
+            Single step on spinbox.
+        value: float
+            Initial value.
+        """
+        super().__init__()
+
+        self.setDecimals(FLOAT_SPINBOX_DECIMALS)
+
+        self.setKeyboardTracking(False)
+
+        self.setMinimum(minimum)
+        self.setMaximum(maximum)
+        self.setSingleStep(step)
+        self.setValue(value)
+
+    def reset_connections(self):
+        """Reset all connections to custom slots."""
+        for slot_func in (self.snap_to_value, self.search_by_function):
+            for signal_func in (self.valueChanged, self.textChanged):
+                try:
+                    signal_func.disconnect(slot_func)
+                except TypeError:
+                    return
```

Do you really want to return on the first fail rather than just suppress and continue just in case you're left with half-formed objects?

---

Yes probably best to forge on with a simple message when raising `TypeError`:
```
    def reset_connections(self):
        """Reset all connections to custom slots."""
        for slot_func in (self.snap_to_value, self.search_by_function):
            for signal_func in (self.valueChanged, self.textChanged):
                try:
                    signal_func.disconnect(slot_func)
                except TypeError:
                    print(f"Callable {signal_func} not a recognised signal in {type(self)}. No slots were disconnected.")
```